drivers/mmc/core/sd_ops.c

```diff
@@ -460,6 +464,26 @@ static int mmc_sd_cmdq_switch(struct mmc_card *card, bool enable)
 	/* Performance enhancement register byte 262 controls command queueing */
 	err = mmc_sd_write_ext_reg(card, card->ext_perf.fno, card->ext_perf.page,
 				   card->ext_perf.offset + 262, reg);
+	if (err)
+		goto out;
+
+	/* Poll the register - cards may have a lazy init/deinit sequence. */
+	timeout = ktime_add_ms(ktime_get(), 10);
+	while (1) {
+		timedout = ktime_after(ktime_get(), timeout);
```

Comment:
Do you have to check for timeout here? It's counterintuitive, and will result in one extra time after the timeout has elapsed. If intentional, it warrants a comment.

Reply:
The timeout length is arbitrary, but the boolean can be removed - no real need to store it separately.

Reply:
Good - it will remove the feeling that something clever might be happening.